wgsl/index.bs

```diff
@@ -1366,86 +1375,360 @@ Therefore, a data value must not be placed in the padding at the end of a struct
+        //   2. Then apply the '[1]' array index phrase, yielding a reference to
+        //      the storage for second component of the vector referenced from
+        //      the previous step.  The result has type ref<function,f32>.
+        // Evaluating the right-hand side of the assignment yields the f32 value 2.0.
+        // Store the f32 value 2.0 into the storage memory locations referenced by uv[1].
+        uv[1] = 2.0;
+
+        var m: mat3x2<f32>;
+        // When evaluating 'm[2]':
+        // 1. First evaluate 'm', yielding a reference to the storage for
+        //    the 'm' variable. The result has type ref<function,mat3x2<f32>>.
+        // 2. Then apply the '[2]' array index phrase, yielding a reference to
+        //    the storage for the third column vector pointed at by the reference
+        //    value from the previous step.
+        //    Therefore the 'm[2]' expression has type ref<function,vec2<f32>>.
+        // The 'const' declaration is for type vec2<f32>, so the declaration
+        // statement requires the initializer to be of type vec2<f32>.
+        // The Load Rule applies (because no other type rule can apply), and
+        // the evaluation of the initializer yields the vec2<f32> value loaded
+        // from the memory locations referenced by m[2] at the time the declaration
+        // is executed.
+        const p_m_col2: vec2<f32> = m[2];
+
+        var A: array<i32,5>;
+        // When evaluating 'A[4]'
+        // 1. First evaluate 'A', yielding a reference to the storage for
+        //    the 'A' variable. The result has type ptr<function,array<i32,5>>.
+        // 2. Then apply the '[4]' array index phrase, yielding a reference to
+        //    the storage for the fifth element of the array referenced by
+        //    the reference value from the previous step.
+        //    The result value has type ref<function,i32>.
+        // The const declaration requires the right-hand-side to be of type i32.
+        // Apply the Load Rule, so the initializer evaluates to the i32 value loaded
+        // from the memory reference computed in step 2, at the time the the
+        // declaration is executed.
+        const p_A_4: ptr<function,i32> = A[4];
+
+        // When evaluating 'person.weight'
+        // 1. First evaluate 'person', yielding a reference to the storage for
+        //    the 'person' variable declared at module scope.
+        //    The result has type ref<private,S>.
+        // 2. Then apply the '.weight' member access phrase, yielding a reference to
+        //    the storage for the second member of the memory referenced by
+        //    the reference value from the previous step.
+        //    The result has type ref<private,f32>.
+        // The const declaration requires the right-hand-side to be of type f32.
+        // Apply the Load Rule, so the initializer evaluates to the i32 value loaded
```

Comment:
Similarly to above, `person.weight` evaluates to an f32 reference. The `&` operator converts the reference to its corresponding pointer value.